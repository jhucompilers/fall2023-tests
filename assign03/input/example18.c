struct Foo {
  int x;
  char y;
};

int readint(void);

int main(void) {
  int result;
  result = 0;
  int f;
  f = readint();
  if (f > 0) {
    struct Foo f;
    f.x = readint();
    f.y = '$';
    while (f.x > 0) {
      if ((f.x / 2) * 2 == f.x) {
        const char *f;
        f = "hello";
        result = f[0];
      } else {
        unsigned long f;
        f = 160000111;
        result = f / 455;
      }
      f.x = f.x - 1;
    }
    f.x = readint();
  }
  return result;
}
