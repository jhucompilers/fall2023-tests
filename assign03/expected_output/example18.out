0|struct Foo|type|struct Foo {}
1|x|variable|int
1|y|variable|char
0|readint|function|function () returning int
0|main|function|function () returning int
2|result|variable|int
2|f|variable|int
3|f|variable|struct Foo {int, char}
5|f|variable|pointer to const char
5|f|variable|unsigned long
